core/src/main/java/org/infinispan/topology/ClusterCacheStatus.java

```diff
@@ -172,6 +184,21 @@ private boolean addMember(Address joiner, CacheJoinInfo joinInfo) {
             return false;
          }
 
+         // Validate if the member is allowed to join
+         if (persistentState.isPresent()) {
```

Comment:
All of the original members need to be part of the new cluster. All of them
persist state locally. If you lose local state then the cluster won't start
and you need to perform manual repair
On 4 Apr 2016 16:18, "Dan Berindei" <notifications@github.com> wrote:

> In core/src/main/java/org/infinispan/topology/ClusterCacheStatus.java
> <https://github.com/infinispan/infinispan/pull/4141#discussion_r58382325>:
>
> > @@ -172,6 +184,21 @@ private boolean addMember(Address joiner, CacheJoinInfo joinInfo) {
> >              return false;
> >           }
> >
> > +         // Validate if the member is allowed to join
> > +         if (persistentState.isPresent()) {
>
> So the cache's persistent state must already exist on the coordinator?
> What if one of the nodes' disk is lost, and that happens to be the first
> node to start?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/infinispan/infinispan/pull/4141/files/ad6d83239ef0b581c1fc83315e347cb36b2573ee#r58382325>
>